pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -268,8 +268,10 @@ func checkIfDaemonPlistFileExists() error {
 	if err := launchd.CheckPlist(*daemonConfig); err != nil {
 		return err
 	}
-	if !launchd.AgentRunning(daemonConfig.Label) {
-		return fmt.Errorf("launchd agent '%s' is not running", daemonConfig.Label)
+	if !isDaemonRunningWithReleasedVersion() {
+		if !launchd.AgentRunning(daemonConfig.Label) {
+			return fmt.Errorf("launchd agent '%s' is not running", daemonConfig.Label)
+		}
```

Comment:
What about writing it like this? This seems clearer regarding what the code wants to do.

```
if olderDaemonVersionRunning() {
   if err := killDaemonProcess(); err != nil {
       return err
   }
}
if !launchd.AgentRunning(daemonConfig.Label) {
			return fmt.Errorf("launchd agent '%s' is not running", daemonConfig.Label)
}
```

Reply:
This will be also easy as once #3099 in.

Reply:
@cfergeau It is now suggested way.

Reply:
Anjan had a good point though that the killing fits better in a `fix` method rather than in `check`.